### DISCUSSION TITLE
raw feedback sound play to i2s when A2DP Sink is connected

### DISCUSSION THREAD

**Discussion by GuidoLec** [pschatzmann/ESP32-A2DP#395] [Q&A] [answered]
Thank you for your wonderful work. I managed to make a lot of nasty features using it. Now I have a full working BT receiver with PCM5102 DAC. It also has static pin code verification thanks to recompiling ESP-IDF and with BluetoothA2DPSinkQueued plus 32-bit and some volume set tweak to soften changes it now sounds perfect. 

Now I'd like to play a feedback raw sound when a device connects to BT Sink. Do you think it is possible to send 8 bit unsigned raw using size_t BluetoothA2DPSink::i2s_write_data(const uint8_t* data, size_t item_size) function?

thanks in advance

**Comment by pschatzmann** [comment, marked as answer]
No, you will need to write the signal to I2S. To do this properly you should deactivate the I2S management of the A2DP library and use your own logic. 

The AudioTools library can help you to do this....

You can get some inspiration from the following examples:
https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-memory_mp3_short-i2s
https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-basic-api/basic-a2dp-i2s/basic-a2dp-i2s.ino
